src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java

```diff
@@ -352,23 +352,20 @@ void update(
      * Finds the {@link RTPEncodingDesc} that corresponds to the packet that is
      * specified in the buffer passed in as an argument.
      *
-     * @param buf the byte array that holds the RTP packet.
-     * @param off the offset in the byte array where the actual data starts
-     * @param len the length of the actual data
+     * @param pkt the packet to match.
      * @return the {@link RTPEncodingDesc} that corresponds to the packet that is
      * specified in the buffer passed in as an argument, or null.
      */
-    RTPEncodingDesc findRTPEncodingDesc(byte[] buf, int off, int len)
+    RTPEncodingDesc findRTPEncodingDesc(RawPacket pkt)
     {
-        if (buf == null || buf.length < off + len
-            || ArrayUtils.isNullOrEmpty(rtpEncodings))
+        if (ArrayUtils.isNullOrEmpty(rtpEncodings))
```

Comment:
should we add a check for `!pkt.isInvalid()`?

Reply:
Sure. I would add it either on top of the chain (so that it triggers early) or at the bottom (so that it is close to the code which actually needs it). Or maybe even both, but I don't like it here in the middle. Any preference?

Reply:
at the top of the chain sounds good.  maybe tweak the javadoc to show that it assumes/requires the `RawPacket` is non-null & valid in the places down the chain?